ipn/ipnlocal/local.go

```diff
@@ -7456,3 +7467,41 @@ func maybeUsernameOf(actor ipnauth.Actor) string {
 	}
 	return username
 }
+
+// VIPServices returns the list of tailnet services that this node
+// is serving as a destination for.
+func (b *LocalBackend) VIPServices() []*tailcfg.VIPService {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	return b.vipServicesFromPrefs(b.pm.CurrentPrefs())
+}
+
+func (b *LocalBackend) vipServicesFromPrefs(prefs ipn.PrefsView) []*tailcfg.VIPService {
+	// keyed by service name
+	services := make(map[string]*tailcfg.VIPService)
```

Comment:
you can keep this nil for (without make-ing it) and lazily make it with `mak.Set` if you want